divfree/TPZCompElHCurlNoGrads.cpp

```diff
@@ -0,0 +1,369 @@
+      //   switch(TSHAPE::Type(TSHAPE::NCornerNodes+nEdges + iFace)){
+      //   case ETriangle://triangular face
+      //     count +=  (order - 1) * ( order - 2)/2;
+      //     break;
+      //   case EQuadrilateral://quadrilateral face
+      //     //we need the functions of k+1
+      //     count +=  order * order;
+      //     break;
+      //   default:
+      //     PZError<<__PRETTY_FUNCTION__<<" error."<<std::endl;
+      //     DebugStop();
+      //   }
+      // }
+
+      // const int nVkf = count;
+      // //now we count the interior bubble functions
+            
+      // //number of H1 funcs
+      // const auto orderH1 = TSHAPE::Type() == ECube ?
+      //   order +1 : order;
+            
+      // const auto nFuncsH1 = TSHAPE::NConnectShapeF(side, orderH1);
+      // if constexpr (TSHAPE::Type() == ECube) {
+      //   //shapeorders[k] contains the polynomial orders
+      //   //in xi, eta and zeta for each of the internal funcs
+      //   TPZGenMatrix<int> shapeorders(nFuncsH1,3);
+      //   //not really used since we are interested in internal funcs
+      //   TPZManVector<int64_t,0> idvec(0);
+      //   TSHAPE::SideShapeOrder(side,idvec,orderH1,shapeorders);              
+      //   /*for compatibility, we need the spaces
+      //     Q^{k,k+1,k+1} \times Q^{k+1,k,k+1} \times Q^{k+1,k+1,k}
+      //     I am supposing that #funcs in xi-dir is equal to 
+      //     #funcs in eta-dir and in zeta-dir. So I will just
+      //     see how many we will have in xi-dir after skipping
+      //     internals that were included in the k-1 set of funcs*/
+      //   int countInt = 0;
+      //   for(int ifunc = 0; ifunc < nFuncsH1; ifunc++){
+      //     if(shapeorders(ifunc,0) <= order) {countInt++;}
+      //   }
+      //   count += 3* countInt;
+
+      //   const auto nVki = count - nVkf;
+      //   // std::stringstream sout;
+      //   // sout << __PRETTY_FUNCTION__<<'\n'
+      //   //      <<"\tside "<<side<<"\tcon "<<icon<<"\torder "<<order<<'\n'
+      //   //      <<"\tnfuncs "<<count<<"\tnvkf "<<nVkf<<"\tnvki "<<nVki<<'\n';
+      //   // std::cout<<sout.str()<<std::flush;
+      // }
+      // else{
+      //   count += 3 * nFuncsH1;
+      // }
+      // return count;
+    }
+  }();
+  return nShapeF;
+}
+
+template<class TSHAPE>
+void TPZCompElHCurlNoGrads<TSHAPE>::ComputeShape(TPZMaterialData &data,
+                                                 TPZFMatrix<REAL> &phiHCurl)
+{
+  constexpr auto dim = TSHAPE::Dimension;
+  constexpr auto nNodes = TSHAPE::NCornerNodes;
+  constexpr auto nConnects = TSHAPE::NSides - nNodes;
+  const auto nFaces = TSHAPE::NumSides(2);
+  const auto nEdges = TSHAPE::NumSides(1);
+  
+  //unfiltered shape functions count for each connect
+  TPZManVector<int,nConnects> firstHCurlFunc(nConnects,0);
+  for(auto icon = 1; icon < nConnects; icon++){
+    const auto conorder = this->ConnectOrder(icon);
+    firstHCurlFunc[icon] = firstHCurlFunc[icon-1] +
```

Comment:
If the connect order is not zero, then we have a bug?

Reply:
I am supposing that the connect orders should be valid orders of the "traditional" HCurl element from which we will filter the functions, therefore this would be the "_leave the connect order to whatever value and change only the number of shape functions associated with the connect_" approach.